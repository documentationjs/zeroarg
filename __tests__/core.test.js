var core = require("../core");

test("core", () =>
  core(() => {
    /**
     * Add two
     */
    function add(a, b) {
      return a + b;
    }
  }).then(yargsConfig => {
    expect(yargsConfig).toMatchSnapshot();
  }));

test("string types", () =>
  core(() => {
    /**
     * Add two
     * @param {string} a
     * @param {string} b
     */
    function add(a, b) {
      return a + b;
    }
  }).then(yargsConfig => {
    expect(yargsConfig).toMatchSnapshot();
  }));

test("non-nullable types", () =>
  core(() => {
    /**
     * Add two
     * @param {string!} a
     * @param {string!} b
     */
    function add(a, b) {
      return a + b;
    }
  }).then(yargsConfig => {
    expect(yargsConfig).toMatchSnapshot();
  }));

test("options", () =>
  core(() => {
    /**
     * Add two
     * @param {string!} a
     * @param {string!} b
     * @param {Object} options
     * @param {string} options.how
     */
    function add(a, b, { how }) {
      return a + b;
    }
  }).then(yargsConfig => {
    expect(yargsConfig).toMatchSnapshot();
  }));

test("options with choices", () =>
  core(() => {
    /**
     * Add two
     * @param {string!} a
     * @param {string!} b
     * @param {Object} options
     * @param {add|divide|multiply} options.how
     */
    function add(a, b, { how }) {
      return a + b;
    }
  }).then(yargsConfig => {
    expect(yargsConfig).toMatchSnapshot();
  }));

test("fail with option in middle", () =>
  core(() => {
    /**
     * Add two
     * @param {string!} a
     * @param {Object} options
     * @param {add|divide|multiply} options.how
     * @param {string!} b
     */
    function add(a, b, { how }) {
      return a + b;
    }
  }).catch(err => {
    expect(err).toMatchSnapshot();
  }));

test("variadic in middle", () =>
  core(() => {
    /**
     * Add two
     * @param {string!} a
     * @param {Array<string>} b
     * @param {string!} c
     */
    function add(a, b, c) {
      return a + b;
    }
  }).catch(err => {
    expect(err).toMatchSnapshot();
  }));
