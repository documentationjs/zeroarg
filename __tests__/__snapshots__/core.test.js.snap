// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core 1`] = `
Object {
  "command": "run [a] [b]",
  "epilogue": "
Add two
",
  "options": Object {},
  "positionalArgumentNames": Array [
    "a",
    "b",
  ],
}
`;

exports[`fail with option in middle 1`] = `[Error: Options must be specified last]`;

exports[`non-nullable types 1`] = `
Object {
  "command": "run <a> <b>",
  "epilogue": "
Add two
",
  "options": Object {},
  "positionalArgumentNames": Array [
    "a",
    "b",
  ],
}
`;

exports[`options 1`] = `
Object {
  "command": "run <a> <b>",
  "epilogue": "
Add two
",
  "options": Object {
    "how": Object {
      "describe": "...",
      "type": "string",
    },
  },
  "positionalArgumentNames": Array [
    "a",
    "b",
  ],
}
`;

exports[`options with choices 1`] = `
Object {
  "command": "run <a> <b>",
  "epilogue": "
Add two
",
  "options": Object {
    "how": Object {
      "choices": Array [
        "add",
        "divide",
        "multiply",
      ],
      "describe": "...",
      "type": "string",
    },
  },
  "positionalArgumentNames": Array [
    "a",
    "b",
  ],
}
`;

exports[`string types 1`] = `
Object {
  "command": "run [a] [b]",
  "epilogue": "
Add two
",
  "options": Object {},
  "positionalArgumentNames": Array [
    "a",
    "b",
  ],
}
`;

exports[`variadic in middle 1`] = `[Error: A variadic positional argument must be the last positional argument]`;
